src/runtime-rs/crates/hypervisor/src/device/device_manager.rs

```diff
@@ -76,13 +76,11 @@ impl DeviceManager {
     }
 
     pub async fn new_device(&mut self, device_config: &DeviceConfig) -> Result<String> {
-        let device_id = if let Some(dev) = self.find_device(device_config).await {
-            dev
-        } else {
-            self.create_device(device_config)
-                .await
-                .context("failed to create device")?
-        };
+        let device_id = self
```

Comment:
If the new_device was the simple wrapper of the create_device, why we need those two trait, how about remove one of them?

Reply:
Thx, only new_device will be enough.